flags/templatetags/feature_flags.py

```diff
@@ -15,13 +15,18 @@
     simple_tag = register.assignment_tag
 
 
-@simple_tag(takes_context=True)
-def flag_enabled(context, flag_name):
-    request = context.get('request', None)
-    return base_flag_enabled(flag_name, request=request)
+def _flag_fn(fn, name):
+    @simple_tag(takes_context=True, name=name)
+    def wrapped(context, flag_name, request=None, **kwargs):
+        if request is None:
+            request = context.get('request')
+        return fn(flag_name, request=request, **kwargs)
 
+    return wrapped
```

Comment:
I get trying to not repeat wrapping the function in both `flag_enabled` and `flag_disabled` and getting the `request`, but I honestly think this ends up a lot more complicated to read than just defining the functions normally.

If the intent here was to avoid writing the `if request is None` twice, then maybe a simpler way to write it would be

```python
request = request or context.get('request')
```